src/object-properties.js

```diff
@@ -27,13 +31,60 @@ const properties = {
    */
   navigationMode: 'free',
   /**
+   * Resize and pan chart when a node's list of children is expanded
    * @type {boolean}
    */
   resizeOnExpand: false,
   /**
-   * @type {object}
+   * Holds chart styling
+   * @type {Style}
    */
   style: {},
 };
 
+/**
+ * Holds styling options
+ * @typedef {object} Style
+ * @property {FontColor=} fontColor - Color of the text, by default #484848
+ * @property {Background=} backgroundColor - Color of the background, by default #e6e6e6
+ * @property {Border=} border - Styling for border, by default #737373
+ */
+
+/**
+ * @typedef {object} FontColor
+ * @property {('auto'|'colorPicker'|'byExpression')=} colorType - How the font color is defined, defaults to auto
```

Comment:
For properties with primitive values and where the value is optional. You can specify a default with something like this I believe:
`@property {('auto'|'colorPicker'|'byExpression')} [colorType='auto']`

That would should generate a default value in the spec as well so no need to type the default value in the property description. For the case with objects as props I'll have to get back to you on how to best define that.

Reply:
That is really useful. Would it still be considered a default value even if there is no real default value on the properties per se, but handled with a fallback value in the styling logic? Asking since I wonder if this would work for the table as well

Reply:
I would say the default value is there to indict to the user how the behaviour will be, given that they do not provide a value. So it should not matter how it's implemented. As long as the default value results in the same behaviour as when they do set an explicit value.